src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -517,6 +517,9 @@ await _solutionLoadedEvent.WaitAsync()
                             {
                                 if (isAllProjectsNominated)
                                 {
+                                    // Counterfactual logging for bulk file restore coordination scenario
+                                    Common.TelemetryActivity.EmitTelemetryEvent(new CpsBulkFileRestoreCoordinationEvent());
```

Comment:
Not sure why we need this event in the first place. 

The vs/nuget/restoreinformation event already has the equivalent information, ie, `Implicit` restore source, and not solution load. 

Can we use that instead of the new event?

They have the same purpose.

Reply:
Chatted with @jebriede. 

The event has to be fired before.